applications/DG-Max/Programs/DGMaxSource.cpp

```diff
@@ -123,6 +126,27 @@ class TestingProblem : public HarmonicProblem<dim> {
         Base::PhysicalFace<dim>& face) const override {
         return {};
     }
+
+    DGMax::BoundaryConditionType getBoundaryConditionType(
+        const Base::Face& face) const final {
+        pface.setFace(&face);
+        pface.setPointReference(face.getReferenceGeometry()->getCenter());
+        if (face.isInternal()) {
+            return DGMax::BoundaryConditionType::INTERNAL;
+        }
+
+        LinearAlgebra::SmallVector<dim> normal = pface.getUnitNormalVector();
+
+        double nx = std::abs(std::abs(normal[0]) - 1.0);
+        if (nx < 1e-8) {
+            return DGMax::BoundaryConditionType::DIRICHLET;
+        } else {
+            return DGMax::BoundaryConditionType::NEUMANN;
+        }
+    }
+
+   private:
+    mutable Base::PhysicalFace<dim> pface;
```

Comment:
?

Reply:
Eliminated its need. Was much easier than I though (should have looked earlier).